c2rust-analyze/src/rewrite/expr/hir_op.rs

```diff
@@ -221,6 +238,12 @@ impl<'a, 'tcx> intravisit::Visitor<'tcx> for HirRewriteVisitor<'a, 'tcx> {
     fn visit_expr(&mut self, ex: &'tcx hir::Expr<'tcx>) {
         let mut hir_rw = Rewrite::Identity;
 
+        // This span will be used to apply the actual rewrite.
+        // To prevent attempts to rewrite macros in their definition
+        // location (e.g. `std::ptr::addr_of`) we instead specify that
+        // the rewrite should occur at the callsite
+        let callsite_span = ex.span.source_callsite();
```

Comment:
Always looking through one level of macro expansion is probably not the right answer in all situations, but should work for the code we get out of `c2rust-transpile` at least.